index.js

```diff
@@ -125,16 +193,42 @@ module.exports = () => {
      * added / dropped peers.
      */
     _sendMessage () {
-      var localAdded = Object.keys(this._localAddedPeers).slice(0, PEX_MAX_PEERS)
-      var localDropped = Object.keys(this._localDroppedPeers).slice(0, PEX_MAX_PEERS)
-
-      var added = Buffer.concat(localAdded.map(string2compact))
-      var dropped = Buffer.concat(localDropped.map(string2compact))
-
-      var addedFlags = Buffer.concat(localAdded.map(() => {
-        // TODO: support flags
-        return Buffer.from([0])
-      }))
+      const localAdded = Object.keys(this._localAddedPeers).slice(0, PEX_MAX_PEERS)
+      const localDropped = Object.keys(this._localDroppedPeers).slice(0, PEX_MAX_PEERS)
+
+      const _isIPv4 = (peers, addr) => peers[addr].ip === 4
+      const _isIPv6 = (peers, addr) => peers[addr].ip === 6
```

Comment:
maybe this would be simpler if there were a separate `_localAddedPeers` for ipv6

Reply:
The good part of having it in the same object is that when we have to choose only 50 peers, there are in the same "bucket" so you don't have to be taking them from different places 😅
Giving that the filtering seems quite cheep, I seen more "pros" than "cons". What do you think?

Reply:
hm, yea

It might be more readable if you split the localAdded and localDropped into ipv4 and ipv6 local variables though, instead of filtering multiple times below. Not sure. It's just kind of hard to parse

Reply:
I've refactored this part, hope it is more clear now 😅